pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -801,23 +777,20 @@ pub mod pallet {
 				Error::<T>::RouterNotFound
 			);
 
-			let nonce = <OutboundMessageNonceStore<T>>::try_mutate(|n| {
-				n.ensure_add_assign(T::OutboundMessageNonce::one())?;
-				Ok::<T::OutboundMessageNonce, DispatchError>(*n)
-			})?;
-
-			OutboundMessageQueue::<T>::insert(
-				nonce,
-				(destination.clone(), T::Sender::get(), message.clone()),
-			);
-
-			Self::deposit_event(Event::OutboundMessageSubmitted {
-				sender: T::Sender::get(),
-				domain: destination,
-				message,
-			});
+			match PackedMessage::<T>::get((&sender, &destination)) {
+				Some(packed) => {
+					PackedMessage::<T>::insert((sender, destination), packed.pack(message)?)
+				}
+				None => Self::enqueue_message(destination, message)?,
+			}
 
 			Ok(())
 		}
+
+		fn weight() -> Weight {
+			T::DbWeight::get()
+				.reads_writes(3, 4)
+				.saturating_add(Weight::from_parts(0, T::Message::max_encoded_len() as u64))
+		}
```

Comment:
Here we compute the weight needed to enqueue a message the could be a batch. This size is different to the used when sending the message, which use the serialized sized.